app/upgrades/v17/upgrades_test.go

```diff
@@ -0,0 +1,142 @@
+package v17_test
```

Comment:
Can you implement the test from this file into your implementation? https://github.com/osmosis-labs/osmosis/blob/e0ef3fd28ede4784f0f002270a128a817c8be908/app/upgrades/v17/upgrades_test.go

This would also test my above question about the pools that get created in the upgrade handler if they get a twap record created when they are created in the upgrade handler

Reply:
hmm i'm thinking about waiting to get it merged and then rebasing, so that it is a little easier to review here. But i'm pretty confident the response to the above question